src/features/resurrection/hooks/useResurrection.ts

```diff
@@ -57,7 +57,13 @@ export function useResurrection(sarcoId: string, recipientPrivateKey: string) {
       const outerLayerBuffer = Buffer.from(outerLayer);
 
       // Convert the shards from their hex strings to Uint8Array
-      const unencryptedShards = archaeologists.map(a => Buffer.from(arrayify(a.unencryptedShard)));
+      const unencryptedShards = archaeologists
+        .map(a => {
+          if (arrayify(a.unencryptedShard).length > 0) {
```

Comment:
Minor nitpick but, how about a `const arrayifiedShard = arrayify(a.unencryptedShard)` before the `if` so `arrayify` is called just once?

Reply:
Good suggestion. Should have done that originally.